src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs

```diff
@@ -557,6 +557,132 @@ public int Length
             }
         }
 
+        /// <summary>
+        /// EnumerateChunks returns ChunkEnumerable that follows the IEnumerable pattern and
+        /// thus can be used in a C# 'foreach' statemens to retreive the data in the StringBuilder
+        /// as chunks (ReadOnlyMemory) of characters.  An example use is:
+        /// 
+        ///      foreach (ReadOnlyMemory<char> chunk in sb.EnumerateChunks())
+        ///      {  /* operate on chars using 'chunk' */   }
+        ///      
+        /// </summary>
+        public ChunkEnumerable EnumerateChunks() => new ChunkEnumerable(this);
+
+        /// <summary>
+        /// ChunkEnumerable supports the IEnumerable pattern so foreach works (see EnumerateChunks)  
```

Comment:
Worth explaining here why you didn't make String builder implement IEnumerable? (Now you moved off Span, you could?)

Reply:
There is not a strong barrier to implementing IEnumerable.    The main reason for not doing so is that it does incur some cost in complexity (more methods (that mostly never get called) and testing, and does not seem to add important value.  Its main potential value is that you could pass it to APIs that take a IEnumerable, but that definitely would decrease the performance (you would end up boxing the struct), and chunks are really not a semantic meaningful quantity (it is note the natural thing you would do Linq over).   I don't we should be encouraging using Linq over chunks.  

So while it may have some value (which is why this question raised in https://github.com/dotnet/corefx/issues/29770), given that we could add it later compatibly, it is lower risk not to include it.

I hesitate to modify this comment however, as it is public facing documentation (and probably should avoid design discussions (which are better in PRs and Issues).